src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -484,46 +484,44 @@ ImmutableArray<IMemberInitializer> IObjectCreationExpression.MemberInitializers
 
         private class FieldInitializer : IFieldInitializer
         {
-            public FieldInitializer(SyntaxNode syntax, IFieldSymbol field, IExpression value)
+            public FieldInitializer(SyntaxNode syntax, IFieldSymbol initializedField, IExpression value)
             {
                 this.Syntax = syntax;
-                this.Field = field;
+                this.InitializedField = initializedField;
                 this.Value = value;
             }
 
-            public IFieldSymbol Field { get; }
+            public IFieldSymbol InitializedField { get; }
 
-            MemberInitializerKind IMemberInitializer.MemberInitializerKind => MemberInitializerKind.Field;
+            public ImmutableArray<IFieldSymbol> InitializedFields => ImmutableArray.Create(this.InitializedField);
```

Comment:
Would it make sense to cache the array instead and return the same array every time. InitializedField property can simply return first element from the array.